src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java

```diff
@@ -629,35 +650,34 @@ protected static String parseJavadocMemory( String memory )
             throw new IllegalArgumentException( "The memory could not be null." );
         }
 
-        Pattern p = Pattern.compile( "^\\s*(\\d+)\\s*?\\s*$" );
+        Pattern p = PARSE_JAVADOC_MEMORY_PATTERN_0;
         Matcher m = p.matcher( memory );
         if ( m.matches() )
         {
             return m.group( 1 ) + "m";
         }
-
-        p = Pattern.compile( "^\\s*(\\d+)\\s*k(b)?\\s*$", Pattern.CASE_INSENSITIVE );
+        p = PARSE_JAVADOC_MEMORY_PATTERN_1;
         m = p.matcher( memory );
         if ( m.matches() )
         {
             return m.group( 1 ) + "k";
         }
 
-        p = Pattern.compile( "^\\s*(\\d+)\\s*m(b)?\\s*$", Pattern.CASE_INSENSITIVE );
+        p = PARSE_JAVADOC_MEMORY_PATTERN_2;

```

Comment:
can you just use the constants here rather than assigning and reusing a local variable?

Reply:
@elharo done.